src/test/java/ir/sahab/uncaughtexceptionrule/UncaughtExceptionExtensionTest.java

```diff
@@ -0,0 +1,61 @@
+package ir.sahab.uncaughtexceptionrule;
+
+import org.apache.log4j.BasicConfigurator;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class UncaughtExceptionExtensionTest {
+
+    @RegisterExtension
+    UncaughtExceptionExtension uncaughtExceptionExtension = new UncaughtExceptionExtension();
+
+    @BeforeAll
+    static void setup() {
+        BasicConfigurator.configure();
+    }
+
+    @Test
+    void testAssertForUnhandledExceptions() throws InterruptedException {
+        Thread t = new Thread(() -> {
+            throw new ArithmeticException();
+        });
+        t.start();
+        t.join();
+
+        assertNotNull(uncaughtExceptionExtension.getException());
+        assertTrue(uncaughtExceptionExtension.getException() instanceof ArithmeticException);
+        // Clear the exception for passing the running unit test
+        uncaughtExceptionExtension.clearException();
+    }
+
+    /*
+     * This test will fail to show that our extension can catch exceptions from other threads.
+     */
+    @Test
+    @Disabled
+    void testFailureForUnhandledException() throws InterruptedException {
+        Thread t = new Thread(() -> {
+            throw new ArithmeticException();
+        });
+        t.start();
+        t.join();
+    }
+
+    // In case of exception in test thread and another thread we should show all exceptions
```

Comment:
Remove this part:
 we should show all exceptions
And instead add:
 , {@link UncaughtExceptionExtension} catch all exceptions.